src/main/java/org/breedinginsight/brapi/v2/services/BrAPITrialService.java

```diff
@@ -300,6 +300,30 @@ public Dataset getDatasetData(Program program, UUID experimentId, UUID datasetId
         log.debug("fetching dataset: " + datasetId + " for experiment: " + experimentId + ".  including stats: " + stats);
         log.debug("fetching observationUnits for dataset: " + datasetId);
         List<BrAPIObservationUnit> datasetOUs = ouDAO.getObservationUnitsForDataset(datasetId.toString(), program);
+
+        //Add years to the addition_info elements
+        Map<String, Integer> studyDbId2Year = new HashMap<>();  // used to prevent the same study from being fetched repeatedly.
+        for ( BrAPIObservationUnit ou: datasetOUs ) {
+            String studyDbId = ou.getStudyDbId();
+
+            if( !studyDbId2Year.containsKey( studyDbId ))  {
+                // Get the Study and extract the year from its Season
+                BrAPIStudy study = studyDAO.getStudyByDbId(studyDbId, program).orElseThrow( () -> new DoesNotExistException(String.format("Study Id '%s' not found.", studyDbId)) );
+                if(study.getSeasons().isEmpty()){
+                    throw new DoesNotExistException(String.format("No Seasons found in Study Id = '%s'.", studyDbId));
+                }
+                String seasonId = study.getSeasons().get(0);
```

Comment:
The "Env Year" in our UI doesn't map cleanly onto BrAPI, because a Study has an array of Seasons, rather than a single Season. Getting the first in the array is a reasonable approach given the mismatch, but I would at least alert Shawn to the issue and get her feedback.

I've attached a screenshot from this [BrAPI UML diagram](https://plantbreeding.github.io/brapi-ontology/uml.html) showing how Season is linked to Study and Observation.

![image](https://github.com/user-attachments/assets/18d4037c-b8e5-442a-b933-c3f8474b564d)

Reply:
I spoke with Shawn. Marking as resolved.